src/AdsClient/Internal/TypeExtensions.cs

```diff
@@ -28,6 +28,8 @@ internal static class TypeExtensions
     [MethodImpl(MethodImplOptions.AggressiveInlining)]
     public static ref byte GetOffset(this ref byte source, int offset)
```

Comment:
Shouldn't offset be `uint` here as well to just avoid the cast at line 33 altogether?

Reply:
That's possible too, but then there's the cast in L25. If the parameter of that method is changed to `uint` too, then the cast will likely be somewhere else.

So I think it's just fine to have it here (once), and not scattered around the code-base.